src/Parser/TypeParser.php

```diff
@@ -381,42 +385,33 @@ public function isHtml(TokenIterator $tokens): bool
 	public function parseGeneric(TokenIterator $tokens, Ast\Type\IdentifierTypeNode $baseType): Ast\Type\GenericTypeNode
 	{
 		$tokens->consumeTokenType(Lexer::TOKEN_OPEN_ANGLE_BRACKET);
-		$tokens->tryConsumeTokenType(Lexer::TOKEN_PHPDOC_EOL);
 
+		$startLine = $baseType->getAttribute(Ast\Attribute::START_LINE);
+		$startIndex = $baseType->getAttribute(Ast\Attribute::START_INDEX);
 		$genericTypes = [];
 		$variances = [];
 
-		[$genericTypes[], $variances[]] = $this->parseGenericTypeArgument($tokens);
-
-		$tokens->tryConsumeTokenType(Lexer::TOKEN_PHPDOC_EOL);
-
-		while ($tokens->tryConsumeTokenType(Lexer::TOKEN_COMMA)) {
+		$isFirst = true;
```

Comment:
Is there a benefit to this change, such as in performance? Personally, I find it more difficult to read this way. Feels like I have to think harder to figure out what's going on here. I understand that's very subjective, just wanted to raise the question :)

Reply:
Previously some code parts were duplicated...

Now the LOC is lower and IMO it is more readable too. In first iteration, we parse/expect no comma, in 2nd+ we parse it with delimiting comma + allow trailing comma.